crates/store/src/state.rs

```diff
@@ -1179,3 +1312,46 @@ async fn load_account_tree(
 
     Ok(AccountTreeWithHistory::new(account_tree, block_number))
 }
+
+/// Loads SMT forest with storage map and vault Merkle paths for all public accounts.
+#[instrument(target = COMPONENT, skip_all, fields(block_num = %block_num))]
+async fn load_smt_forest(
+    db: &mut Db,
+    block_num: BlockNumber,
+) -> Result<InnerForest, StateInitializationError> {
+    use miden_protocol::account::delta::AccountDelta;
+
+    // Skip private accounts - they don't have public state to reconstruct
+    let public_account_ids: Vec<AccountId> = db
+        .select_all_account_commitments()
+        .await?
+        .iter()
+        .filter_map(|(id, _commitment)| if id.has_public_state() { Some(*id) } else { None })
+        .collect();
```

Comment:
This is pretty inefficient because we just need to get a list of accounts with public state, but instead we are loading all account commitments. This is OK for now because we'll replace the initialization code as we migrate to RocksDB-based backends.